module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Logic/HoldsTest.php

```diff
@@ -98,4 +102,266 @@ public function testGetSuppressedLocations(array $configArray, array $expectedLi
+        $result = $this->callMethod($logic, 'formatHoldings', [$holdings]);
+
+        // assert various properties of the result
+        // I need to update the fixture and test to be more readable - remove UUIDs and replace with simple strings
+        $this->assertArrayHasKey('location1_id|Main Library', $result);
+        $this->assertEquals('Main Library', $result['location1_id|Main Library']['location']);
+        $this->assertCount(2, $result['location1_id|Main Library']['items']);
+        $this->assertEquals('02.09.2025', $result['location1_id|Main Library']['items'][0]['duedate']);
+        $this->assertEquals('Circulating', $result['location1_id|Main Library']['items'][0]['loan_type_name']);
+        $this->assertEquals(true, $result['location1_id|Main Library']['items'][0]['is_holdable']);
+        $this->assertEquals(
+            'item_id_1',
+            $result['location1_id|Main Library']['items'][0]['item_id']
+        );
+        $this->assertEquals(
+            'holdings_id_1',
+            $result['location1_id|Main Library']['items'][0]['holdings_id']
+        );
+        $this->assertEquals(
+            'item_id=item_id_1&holdings_id=holdings_id_1&status=Checked+out&hashKey=sample_hash_key',
+            $result['location1_id|Main Library']['items'][0]['link']['query']
+        );
+    }
+
+    /**
+     * Create an availability status for testing
+     *
+     * @param bool   $available   Whether the item is available
+     * @param string $description Status description
+     *
+     * @return AvailabilityStatus
+     */
+    protected function createAvailabilityStatus($available = true, $description = 'Available')
+    {
+        // Use the real AvailabilityStatus class
+        $availability = $available ?
+            \VuFind\ILS\Logic\AvailabilityStatusInterface::STATUS_AVAILABLE :
+            \VuFind\ILS\Logic\AvailabilityStatusInterface::STATUS_UNAVAILABLE;
+
+        return new \VuFind\ILS\Logic\AvailabilityStatus($availability, $description);
+    }
+
+    /**
+     * Test generateHoldings method with different hold types
+     *
+     * @return void
+     */
+    public function testGenerateHoldings(): void
+    {
+        $hmac = $this->createMock(HMAC::class);
+        $hmac->method('generate')->willReturn('test-hash');
+
+        $logic = $this->getHoldsLogic(hmac: $hmac);
+
+        // Example holdings data; lacks 'availability' field
+        $result = $this->getJsonFixture('ils/holdings_formatted_example.json');
+        $availabilityStatus = [
+            [false, 'Checked Out'],
+            [true, 'Available'],
+            [true, 'Available'],
+        ];
+        $i = 0;
+        foreach ($result['holdings'] as &$holding) {
+            // Add 'availability' field using the real class
+            $holding['availability'] = $this->createAvailabilityStatus(
+                $availabilityStatus[$i][0],
+                $availabilityStatus[$i][1]
+            );
+            $i++;
+        }
```

Comment:
Does this simplification work? I think it should as long as the holdings data doesn't have unexpected array keys.

```suggestion
        foreach ($result['holdings'] as $i => &$holding) {
            // Add 'availability' field using the real class
            $holding['availability'] = $this->createAvailabilityStatus(
                $availabilityStatus[$i][0],
                $availabilityStatus[$i][1]
            );
        }
```